app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -154,9 +155,15 @@ public void updateIfNeeded() {
         }
 
         /* Section - GUI */
-        // Enforce TLS.
         Element gui = getGuiElement();
-        changed = setConfigElement(gui, "tls", "true") || changed;
+
+        // Platform-specific: Force REST API and Web UI access to use TLS 1.2 or not.
+        Boolean forceHttps = Constants.osSupportsTLS12();
+        if (!gui.hasAttribute("tls") ||
+                Boolean.parseBoolean(gui.getAttribute("tls")) != forceHttps) {
+            gui.setAttribute("tls", forceHttps ? "true" : "false");
```

Comment:
Are `setConfigElement(gui, ...` (before) and `gui.setAttribute(...` (now) equivalent?

Reply:
No because during fork dev I discovered a problem here as in very early syncthing's it was an xml node and now is an Attribute of the xml tag. So setAttribute is right.